src/Microsoft.SqlTools.ServiceLayer/LanguageServices/Completion/SqlCompletionItem.cs

```diff
@@ -45,7 +46,7 @@ private void Init()
         {
             InsertText = GetCompletionItemInsertName();
             Label = DeclarationTitle;
-            if (StartsWithBracket(TokenText))
+            if (StartsWithBracket(TokenText) || AutoCompleteHelper.IsReservedWord(InsertText))
             {
                 Label = WithBracket(Label);
                 InsertText = WithBracket(InsertText);
```

Comment:
I think for your scenario you should only add the bracket to insert text not the label. because now user see everything in the list without bracket except for objects with reserved word name